cpp/source/deal.II/geometry.templates.h

```diff
@@ -423,6 +441,51 @@ void Geometry<dim>::convert_geometry_database(
     database->put("cathode.dimensions", cathode_dim);
   }
 }
+
+template <int dim>
+void Geometry<dim>::merge_repetition(
+    dealii::distributed::Triangulation<dim> &repetition, double offset)
+{
+  dealii::Tensor<1, dim> shift_vector;
+  shift_vector[0] = offset;
+  dealii::GridTools::shift(shift_vector, repetition);
+  dealii::distributed::Triangulation<dim> tmp_triangulation(
+      _triangulation->get_communicator());
+  dealii::GridGenerator::merge_triangulations(*_triangulation, repetition,
+                                              tmp_triangulation);
+  _triangulation->clear();
+  _triangulation->copy_triangulation(tmp_triangulation);
+}
+
+template <int dim>
+double Geometry<dim>::compute_length(
+    dealii::distributed::Triangulation<dim> const &triangulation,
+    unsigned int dimension,
+    std::unordered_set<dealii::types::material_id> const &excluded_materials)
+{
+  // Since we are working on the coarse mesh, all the processors have the same
+  // mesh and thus there is no need for a reduction or any other communication.
+  double min_pos = std::numeric_limits<double>::max();
+  double max_pos = std::numeric_limits<double>::min();
```

Comment:
NO!!! `auto max_pos = - std::numeric_limits<double>::max();` or `...::lowest()` in C++11.